ui/app/components/secret-edit.js

```diff
@@ -108,60 +88,59 @@ export default Component.extend(FocusOnInsertMixin, WithNavToNearestAncestor, {
     'model',
     'model.id',
     'mode'
-  ),
-  canDeleteSecretMetadata: alias('checkMetadataCapabilities.canDelete'),
-  canUpdateSecretMetadata: alias('checkMetadataCapabilities.canUpdate'),
-  canReadSecretMetadata: alias('checkMetadataCapabilities.canRead'),
+  )
+  checkMetadataCapabilities;
+  @alias('checkMetadataCapabilities.canDelete') canDeleteSecretMetadata;
+  @alias('checkMetadataCapabilities.canUpdate') canUpdateSecretMetadata;
+  @alias('checkMetadataCapabilities.canRead') canReadSecretMetadata;
 
-  requestInFlight: or('model.isLoading', 'model.isReloading', 'model.isSaving'),
+  @or('model.isLoading', 'model.isReloading', 'model.isSaving') requestInFlight;
+  @or('requestInFlight', 'model.isFolder', 'model.flagsIsInvalid') buttonDisabled;
 
-  buttonDisabled: or('requestInFlight', 'model.isFolder', 'model.flagsIsInvalid'),
-
-  modelForData: computed('isV2', 'model', function () {
-    let { model } = this;
+  get modelForData() {
+    let { model } = this.args;
     if (!model) return null;
     return this.isV2 ? model.belongsTo('selectedVersion').value() : model;
-  }),
+  }
 
-  basicModeDisabled: computed('secretDataIsAdvanced', 'showAdvancedMode', function () {
+  get basicModeDisabled() {
     return this.secretDataIsAdvanced || this.showAdvancedMode === false;
-  }),
+  }
 
-  secretDataAsJSON: computed('secretData', 'secretData.[]', function () {
+  get secretDataAsJSON() {
     return this.secretData.toJSON();
-  }),
+  }
 
-  secretDataIsAdvanced: computed('secretData', 'secretData.[]', function () {
+  get secretDataIsAdvanced() {
     return this.secretData.isAdvanced();
-  }),
+  }
 
-  showAdvancedMode: or('secretDataIsAdvanced', 'preferAdvancedEdit'),
+  get showAdvancedMode() {
+    return this.secretDataIsAdvanced || this.args.preferAdvancedEdit;
+  }
```

Comment:
Following up from our zoom conversation that this new getter will cover the scenario where `preferAdvancedEdit` was previously being set in the constructor based on the `isAdvanced` value.